node/network/src/protocol.rs

```diff
@@ -104,19 +123,29 @@ impl Protocol {
                     }
                 }
             }
-            Message::BlockResponse(_request_id, blocks) => {
-                forward_msg(
-                    self.inc_block_tx.clone(),
-                    (peer_id, blocks),
-                );
+            Message::BlockResponse(request_id, blocks) => {
+                match self.requests.read().expect(POISONED_LOCK_ERR).get(&request_id) {
```

Comment:
was thinking if we can make a functional wrapper here, to not repeat `ban_peer` twice. not sure what that would look like though, so ignore if nothing pops up to you.

Reply:
You sort of can do this by composing two predicate functions (one for the type of request, the other for the condition that it should meet), but I think that's a bit too far in terms of refactoring :)